source/extensions/tracers/xray/tracer.cc

```diff
@@ -156,10 +154,9 @@ Tracing::SpanPtr Tracer::startSpan(const std::string& operation_name, Envoy::Sys
   return span_ptr;
 }
 
-XRay::SpanPtr Tracer::createNonSampledSpan() const {
```

Comment:
To preserve constness, we could move setting the span ID to the Span constructor, not clear if the `setId` method is really necessary but preserved it for now since it was added by the original implementer

Reply:
this change is no longer needed, dismissing